### PR TITLE
Zpf application token

### PR DESCRIPTION
**Pull request by JorisHeadease**
This PR adds the ZPF application token request, and it adds an `/viewer/flag` view that can be used to consume this access_token and post a new Flag. This is a flag that works:

```
{
  "resourceType":"Flag",
  "meta":{
    "profile":[
      "http://nictiz.nl/fhir/StructureDefinition/zib-Alert"
    ]
  },
  "language":"nl-NL",
  "extension":[
    {
      "url":"http://hl7.org/fhir/StructureDefinition/flag-priority",
      "valueCodeableConcept":{
        "coding":[
          {
            "system":"http://hl7.org/fhir/flag-priority-code",
            "code":"PH"
          }
        ]
      }
    }
  ],
  "status":"active",
  "category":{
    "coding":[
      {
        "system":"http://loinc.org",
        "code":"74018-3",
        "display":"alert"
      }
    ]
  },
  "code":{
    "coding":[
      {
        "system":"http://snomed.info/sct",
        "code":"392570002",
        "display":"Blood pressure finding (finding)"
      }
    ]
  },
  "subject":{
    "identifier":{
      "system":"http://fhir.nl/fhir/NamingSystem/bsn",
      "value":"999999151"
    }
  },
  "period":{
    "start":"2024-11-11"
  },
  "author":{
    "identifier":{
      "system":"http://hl7.org/fhir/StructureDefinition/oid",
      "value":"007"
    },
    "display":"Test"
  }
}
```

### CODE REVIEW

**Comment by JorisHeadease** [comment]
I've modified this PR to clean up all added swagger-ui stuff from the front-end AND add the application-token request to orca. We'll use this application-token in another PR, when implementing the "zorginzage" logic. This will be a "proper" way of showcasing that ZPF BgZ data can be queried.

@reinkrul Can you have a quick look at the PR, I'll re-use this application token logic in the zorginzage PR later on.

File: frontend/Dockerfile [hunk 1]
```diff
@@ -27,7 +27,7 @@ COPY --from=base /app/node_modules ./node_modules
 COPY --from=base /app/package.json ./package.json
 COPY --from=base /app/next.config.mjs ./next.config.mjs
 COPY --from=base /app/.env.production ./.env.production
-COPY --from=base /app/public ./public
+# COPY --from=base /app/public ./public
```
Comment (reinkrul) [inline frontend/Dockerfile:30]: leftover?

File: orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go [hunk 2]
```diff
@@ -26,6 +26,35 @@ type SecureTokenService interface {
 
 var _ SecureTokenService = &Service{}
 
+func (s *Service) RequestApplicationToken(launchContext LaunchContext) (string, error) {
+
+	assertion, err := s.createSAMLAssertionForApplication(&launchContext)
+	if err != nil {
+		return "Failed to create SAML assertion", err
```
Comment (reinkrul) [inline orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go:33]: just wrap the error?

```suggestion
		return fmt.Errorf("failed to create SAML assertion: %w", err)
```

(also applies to the ones below)

File: orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go [hunk 3]
```diff
@@ -143,6 +172,38 @@ func (s *Service) createSAMLAssertion(launchContext *LaunchContext) (*etree.Elem
...
 }
 
+func (s *Service) createSAMLAssertionForApplication(launchContext *LaunchContext) (*etree.Element, error) {
+
+	assertion, err := s.createSAMLAssertion(launchContext)
+	if err != nil {
+		return nil, err
+	}
+
+	// Replace the PurposeOfUse attribute in the assertion
```
Comment (reinkrul) [inline orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go:182]: wouldn't it be easier to provide these variables to `createSAMLAssertion`?

File: orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go [hunk 4]
```diff
@@ -143,6 +172,38 @@ func (s *Service) createSAMLAssertion(launchContext *LaunchContext) (*etree.Elem
...
+		return nil, fmt.Errorf("PurposeOfUse element not found in assertion")
+	}
+	attrValue.CreateAttr("code", "OPERATIONS")
+
+	// Replace the Role attribute in the assertion
+	roleAttr := assertion.FindElement(".//Attribute[@Name='urn:oasis:names:tc:xacml:2.0:subject:role']//AttributeValue/Role")
+	if roleAttr == nil {
+		return nil, fmt.Errorf("role element not found in assertion")
+	}
+	roleAttr.CreateAttr("code", "182777000")
```
Comment (reinkrul) [inline orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go:194]: same for this

File: orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go [hunk 5]
```diff
@@ -143,6 +172,38 @@ func (s *Service) createSAMLAssertion(launchContext *LaunchContext) (*etree.Elem
...
+
+	//Remove the workflow id if not provided
+	if launchContext.WorkflowId == "" {
```
Comment (reinkrul) [inline orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go:197]: just don't add it in the first place

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go (hunk 5 above)
Comment (JorisHeadease) [reply orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go:197]: We don't want to allow for a HCP token without a workflow though. For the application token, this is "allowed", for example when POST'ing a Flag

**Review by reinkrul — APPROVED** [review]

File: orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go (hunk 5 above)
Comment (JorisHeadease) [reply orchestrator/careplancontributor/applaunch/zorgplatform/handle_create_hcp_rst.go]: Yeah definitely! I've refactored it, only optional for the application token now.

**Comment by JorisHeadease** [comment]
I've modified the PR quite a bit to make it cleaner and have less duplicate code. So I've requested another review to be sure. Especially in areas around:

```
type TokenType string

const (
	HcpTokenType         TokenType = "HCP"
	ApplicationTokenType TokenType = "APPLICATION"
)
```

There might be cleaner ways in golang, like `iota`, more of an enum'ish approach. It looked quite complex for an enum though

**Pull request merged**